srcs/executing/signals.c

```diff
@@ -0,0 +1,52 @@
+#include "minishell.h"
+
+/*
+** 🦕🌴
+**
+**	set_signals
+**
+**	1. CTRL-C
+**		Interrupt the current processes in minishell but NOT exit minishell.
+**	2. CTRL-\
+**		Do nothing. "Quit: 3\n"
+**	3. CTRL-D (EOF)
+**		Already works. Nothing to do.
+**
+**	The way our executor is build, we need to kill all ongoing processes
+**	"manually".
+**
+** 🌴🥥
+*/
+
+static void	signal_handler(int sig_num)
+{
+	if (sig_num == SIGINT)
+	{
+		rl_replace_line("", 0);

```

Comment:
when you press ctrl C in the main process (so in the prompt, in the shell, when there's no cmd executing) a new line is supposed to appear. I think that's what they're doing. You're not supposed to print ^C in this case btw

Reply:
actually you are. Their code prints ^C